packages/invoices/src/OutboundShipment/DetailView/modals/hooks/useDraftOutboundLines.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState, useCallback } from 'react';
```

Comment:
Very little changed here from the previous `useBatchRows` hook, except for obviously moving it to this file. The other change is instead of `BatchRow` which was essentially a `StockLine` (from which we checked afterward comparing with the existing invoice lines to determine if we needed to do anything), returning an array of `DraftOutboundLine` which are a combination of `StockLine` and `InvoiceLine` shapes, so the mapping into `Input` objects to save is a lot easier.